core/autoscaling.go

```diff
@@ -526,12 +566,29 @@ func (a *autoScalingGroup) launchCheapestSpotInstance(
 	}
 	logger.Println("Found on-demand instance", *baseInstance.InstanceId)
 
-	newInstanceType, err := baseInstance.getCheapestCompatibleSpotInstanceType()
-
+	newInstanceType, err = a.getInstanceTypeByTagInASG()
 	if err != nil {
-		logger.Println("No cheaper compatible instance type was found, "+
-			"nothing to do here...", err)
-		return errors.New("no cheaper spot instance found")
+		logger.Println("Errors while trying to fetch tags for instance type...")
+		return errors.New("Errors while trying to fetch tags")
+	}
+
+	// Check option to keep instance type
+	// If we keep the instance type we don't need to calculate the
+	// compatible instance type.
+	// Also ignore if it's already specified in the instance-type tag
+	if newInstanceType == "" {
+		if a.region.conf.KeepInstanceType == "current" {
+			newInstanceType = *baseInstance.InstanceType
+		} else if a.region.conf.KeepInstanceType != "" {
+			newInstanceType = a.region.conf.KeepInstanceType
```

Comment:
I would convert this logic into a match-based filter, allowing the user to pass multiple allowed instance types in a comma-separated list, and should also support using wildcards:

allowed_instance_types: c3.*, m4.*
or 
allowed_instance_types: m4.large
or
allowed_instance_types: same (or current)

This is much like we already do for the filter of allowed regions.